### PR TITLE
if error, output stack trace on new line

### PR DESCRIPTION
**Pull request by ahdinosaur**
hey,

this pull request adds special treatment for errors: if there is an error and it has a stack trace, then output the stack trace on a new line.

i'm not sure if this is a good idea or a terrible idea, but i wanted to see my stack traces so here we are. how do you log errors?

also, if this is an okay change, happy for feedback on the code since it seems awkward. :cat:

cheers!

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -67,7 +67,13 @@ function PinoColada () {
     if (contentLength != null) output.push(formatBundleSize(contentLength))
     if (responseTime != null) output.push(formatLoadTime(responseTime))
 
-    return output.filter(noEmpty).join(' ')
+    var str = output.filter(noEmpty).join(' ')
+
+    // if error, output stack trace on new line
+    var err = obj.err
+    if (err && err.stack) str += nl + err.stack
```
Comment (lrlna) [inline index.js]: ok, so errors get outputed under https://github.com/lrlna/pino-colada/pull/13/files#diff-168726dbe96b3ce427e7fedce31bb0bcR97, and we do print of the stack trace for `fatal`. Are you currently logging those errors as `fatal`, or just `error`? I want to make sure it's not a bug with the way we handle it below, basically.

**Review by lrlna — CHANGES_REQUESTED** [review]

**Comment by lrlna** [comment]
Hiya! Thank you for this :) I had some questions I asked in the review above ^^

File: index.js (hunk 1 above)
Comment (ahdinosaur) [reply index.js]: oh okay cool, i didn't realize that we were printing a stack trace for `fatal`. my errors are coming from [here in `pino-http`](https://github.com/pinojs/pino-http/blob/2778f077ab5b3390af7d01fc550c475ff7b1d925/logger.js#L35-L42), which calls `log.error({ err })`.

what do you think is the best way to handle this use case? should i instead capture the error before it reaches `pino-http` and use `log.fatal(err)`?

Comment (lrlna) [reply index.js]: ahhhhh, getchu. So I think we should print the stack on both `error` and `fatal`. I was going to do that to begin with, but I changed my mind and I am not sure why... SO let's fix it. How about something like this in `formatMessage()`

```js
function formatMessage (obj) {
     if (obj.level === 'error') return chalk.dim.red(formatError(obj))
     if (obj.level === 'trace') return chalk.dim.white(obj.message)
     if (obj.level === 'warn') return chalk.dim.magenta(obj.message)
     if (obj.level === 'debug') return chalk.dim.yellow(obj.message)
     if (obj.level === 'fatal' || obj.level === 'error') return chalk.bgRed(formatError(obj))
     if (obj.level === 'info' || obj.level === 'userlvl') return formatMessageName(obj.message)
   }

  function formatError (obj) {
    return obj.stack ? obj.message + nl + obj.stack : obj.message
  }
```

Comment (ahdinosaur) [reply index.js]: okay sweet. added some more commits: reverted my previous code, used your suggestion (with a minor change), added a way to handle the errors from `pino-http` (when the error is not `obj` but `obj.err`).

**Comment by yoshuawuyts** [comment]
Don't think this patch is a good idea, for people on tiny screens (hi, it me), this can flood the screen and be a real pain for development. In my opinion stack traces should be reserved for crash events only - which corresponds to log level `fatal`.

Also, it's generally recommended with Pino to use `tee(1)` in conjunction with pretty printers for development, and plain output for production. In both those cases you have access to the stack traces already if by any chance you're receiving logs from locations you don't control.

**Comment by lrlna** [comment]
Hey, sorry about the delay on this!

So, we use both pino and pino-colada at work and I've been trying to see how it works with printing out stack traces on error, and I do think it's a bit noisy with them. We've opted out to have a bit more descriptive error messaging and that seems to work for us for now.

```
12:06:11 ⚠️  db:init Schema file /home/scripto/editor/schema/schema-production.sql not found. This might be an error
```

I am going to close this for now, and in the future if it seems like there is more need for this I might reconsider.

**Comment by ahdinosaur** [comment]
oh okay, thanks @lrlna.

**Comment by mcollina** [comment]
@lrlna one thing you might want to add is the top line of the stack, meaning the file and line of code where the error happened.

Another option might be to watch for key-presses, so that pressing `e` will show the last stack that was seen, but that might complicate things too much here.

**Comment by ahdinosaur** [comment]
@lrlna @mcollina if it helps to provide context, my use case for this pull request is using [`pino-http`](https://github.com/pinojs/pino-http), where [request errors are only logged with `log.error`](https://github.com/pinojs/pino-http/blob/08b43ff9eeea558f95930ce2e3b116c752791f74/logger.js#L36-L40) because a request error does not crash the server. as a developer, if a request fails with an error during development, i expect to have enough information to debug _why_ the request failed, where the current situation using `pino-http` and `pino-colada` leaves me with only "request errored". but i guess i'm doing this wrong.

**Comment by ahdinosaur** [comment]
also i totally understand that if you were logging a bunch of harmless errors, then you wouldn't want to see them noisy stack traces. i realize now that at some point i stopped using this patch and [i found another way to handle my use case](https://github.com/dogstack/dogstack-example/blob/fa2c40a76ef648cfaed103bf6cfa6c5881df4d6b/server.js#L52-L55) (using `express`):

```js
// log errors
app.use(function (err, req, res, next) {
  if (err) console.error('error', err)
  next(err)
})
```

**Pull request closed without merging**